Formula/abseil.rb

```diff
@@ -23,15 +23,19 @@ class Abseil < Formula
   fails_with gcc: "5" # C++17
 
   def install
-    mkdir "build" do
-      system "cmake", "..",
-                      *std_cmake_args,
-                      "-DCMAKE_INSTALL_RPATH=#{rpath}",
-                      "-DCMAKE_CXX_STANDARD=17",
-                      "-DBUILD_SHARED_LIBS=ON"
-      system "make"
-      system "make", "install"
-    end
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DCMAKE_INSTALL_RPATH=#{rpath}",
+                    "-DCMAKE_CXX_STANDARD=17",
+                    "-DBUILD_SHARED_LIBS=ON",
+                    "-DABSL_PROPAGATE_CXX_STD=ON",
+                    *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    # Remove bad flags in .pc files.
+    # https://github.com/abseil/abseil-cpp/issues/1408
+    inreplace lib.glob("pkgconfig/absl_random_internal_randen_hwaes{,_impl}.pc"),
+              "-Xarch_x86_64 -Xarch_x86_64 -Xarch_arm64 ", ""
```

Comment:
The `inreplace` does not apply to Linux.

```plaintext
  Error: inreplace failed
  /home/linuxbrew/.linuxbrew/Cellar/abseil/20230125.1/lib/pkgconfig/absl_random_internal_randen_hwaes.pc:
    expected replacement of "-Xarch_x86_64 -Xarch_x86_64 -Xarch_arm64 " with ""
  /home/linuxbrew/.linuxbrew/Cellar/abseil/20230125.1/lib/pkgconfig/absl_random_internal_randen_hwaes_impl.pc:
    expected replacement of "-Xarch_x86_64 -Xarch_x86_64 -Xarch_arm64 " with ""
```

```suggestion
    if OS.mac?
      inreplace lib.glob("pkgconfig/absl_random_internal_randen_hwaes{,_impl}.pc"),
                "-Xarch_x86_64 -Xarch_x86_64 -Xarch_arm64 ", ""
    end
```